src/EthersSafe.ts

```diff
@@ -112,97 +160,135 @@ class EthersSafe implements Safe {
-      throw new Error('Transactions can only be confirmed by Safe owners')
-    }
+  async signTransaction(safeTransaction: SafeTransaction): Promise<void> {
     const txHash = await this.getTransactionHash(safeTransaction)
     const signature = await this.signTransactionHash(txHash)
-    safeTransaction.signatures.set(signature.signer, signature)
+    safeTransaction.addSignature(signature)
+  }
+
+  /**
+   * Approves on-chain a hash using the current signer account.
+   *
+   * @param hash - The hash to approve
+   * @param skipOnChainApproval - TRUE to avoid the Safe transaction to be approved on-chain
+   * @returns The pre-validated signature
+   */
+  async approveTransactionHash(
+    hash: string,
+    skipOnChainApproval?: boolean
+  ): Promise<SafeSignature> {
+    if (!this.#signer) {
+      throw new Error('No signer provided')
+    }
+    const owners = await this.getOwners()
+    if (
+      !owners.find(
+        (owner: string) => this.#signer && areAddressesEqual(owner, this.#signer.address)
+      )
+    ) {
+      throw new Error('Transaction hashes can only be approved by Safe owners')
+    }
+    if (!skipOnChainApproval) {
+      await this.#contract.approveHash(hash)
+    }
+    return generatePreValidatedSignature(this.#signer.address)
   }
 
   /**
-   * Returns the encoding of a Safe transaction.
+   * Returns a list of owners who have approved a specific Safe transaction.
    *
-   * @param transaction - The Safe transaction
-   * @returns The encoding of the Safe transaction
+   * @param txHash - The Safe transaction hash
+   * @returns The list of owners
    */
-  async encodeTransaction(transaction: SafeTransaction): Promise<string> {
-    const encodedTx = await this.#contract.interface.encodeFunctionData('execTransaction', [
-      transaction.data.to,
-      transaction.data.value,
-      transaction.data.data,
-      transaction.data.operation,
-      transaction.data.safeTxGas,
-      transaction.data.baseGas,
-      transaction.data.gasPrice,
-      transaction.data.gasToken,
-      transaction.data.refundReceiver,
-      transaction.encodedSignatures()
-    ])
-    return encodedTx
+  async getOwnersWhoApprovedTx(txHash: string): Promise<string[]> {
+    const owners = await this.getOwners()
+    let ownersWhoApproved: string[] = []
+    for (const owner of owners) {
+      const approved = await this.#contract.approvedHashes(owner, txHash)
+      if (approved.gt(0)) {
+        ownersWhoApproved.push(owner)
+      }
+    }
+    return ownersWhoApproved
   }
 
   /**
    * Executes a Safe transaction.
    *
-   * @param transaction - The Safe transaction to execute
+   * @param safeTransaction - The Safe transaction to execute
    * @param options - Execution configuration options
    * @returns The Safe transaction response
    */
-  async executeTransaction(transaction: SafeTransaction, options?: any): Promise<any> {
+  async executeTransaction(
+    safeTransaction: SafeTransaction,
+    options?: any
```

Comment:
does ethers also not define any type of options?

Reply:
TBD here: https://github.com/gnosis/safe-core-sdk/pull/17